reconcile/utils/terrascript_client.py

```diff
@@ -3277,15 +3459,23 @@ def populate_tf_resource_elasticsearch(self, resource, namespace_info):
             es_values["vpc_options"]['security_group_ids'] = security_group_ids
 
         advanced_options = values.get('advanced_options', None)
+        if advanced_options and \
+           not advanced_options.get('override_main_response_version', None):
+            '''
+            By default terraform expects this to be False. If not set (null),
+            then this results in an 'Update' request on each plan/apply.
+            '''
+            advanced_options['override_main_response_version'] = False
```

Comment:
not sure i understand this part. does it have to be a part of this PR?

Reply:
There is currently an issue that on each plan/apply loop, this change appears:

```
~ advanced_options      = {
            "indices.query.bool.max_clause_count"    = "4000"
          - "override_main_response_version"         = "false" -> null
            "rest.action.multi.allow_explicit_index" = "true"
        }
```

The reason for that is, that terraform tries to set `override_main_response_version` to `null`, because gql returns it in my test data as `None`. However, it won't actually be set to `None`, but to `false` instead, which means this diff will appear on every loop run. By setting it to `false` explicitly, the diff on each subsequent run is clean.

But you are right that this is not necessarily an issue for this ticket. Just sth I stumbled on along the way. Will open a follow up PR afterwards.